### PR TITLE
DSDEGP-1677 DSDEGP-1620 Search backwards in time during recovery.

### PR DESCRIPTION
**Pull request by danxmoran**
### Description

Also, be more lazy in reading documents to recover.

----

### Checklist (never delete this)

Never delete this, it is our record that procedure was followed. If you find that for whatever reason one of the checklist points doesn't apply to your PR, you can leave it unchecked but please add an explanation below.

- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: Ensure that you have added or updated tests
- [ ] **Submitter**: If you're adding/switching libraries or otherwise changing the design, update the [design documentation](https://broadinstitute.atlassian.net/wiki/pages/viewpage.action?pageId=114531509).
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * Add notes on what you've tested
* Review cycle:
  * Team may comment on PR at will
  * **Reviewer assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to reviewer** for further feedback
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Move the JIRA issue to QA once this checklist is completed

### CODE REVIEW

**Comment by pshapiro4broad** [comment]
Would it be possible to test that the code only reads documents from the same day? If not, maybe construct a unit test for the directory filtering code to show that it works as expected.

File: clio-server/src/main/scala/org/broadinstitute/clio/server/dataaccess/PersistenceDAO.scala [hunk 1]
```diff
@@ -168,7 +218,15 @@ object PersistenceDAO {
...
-  val versionFileName = "current-clio-version.txt"
+  val VersionFileName = "current-clio-version.txt"
+
+  /**
+    * Directory depth to naively walk during recovery.
+    *
+    * Should be deep enough to cover date-based partitions of documents in storage,
+    * but not the documents themselves.
+    */
+  val StorageWalkDepth = 3
```
Comment (pshapiro4broad) [inline clio-server/src/main/scala/org/broadinstitute/clio/server/dataaccess/PersistenceDAO.scala]: We should have a comment somewhere that describes the path elements in our Persistence storage, which will make this value less magic. If we have it already, link to it from here, or add it here if we don't have it already.

**Review by pshapiro4broad — APPROVED** [review]

**Comment by danxmoran** [comment]
> Would it be possible to test that the code only reads documents from the same day? If not, maybe construct a unit test for the directory filtering code to show that it works as expected.

[This test](https://github.com/broadinstitute/clio/blob/dm-smarter-recovery-DSDEGP-1677/clio-server/src/test/scala/org/broadinstitute/clio/server/dataaccess/PersistenceDAOSpec.scala#L114) is checking that the backwards search doesn't go back any farther than expected, but it's not worded very well. I'll try to improve it.

File: clio-server/src/main/scala/org/broadinstitute/clio/server/dataaccess/PersistenceDAO.scala (hunk 1 above)
Comment (danxmoran) [reply clio-server/src/main/scala/org/broadinstitute/clio/server/dataaccess/PersistenceDAO.scala]: Done

**Review by tlangs — APPROVED** [review]

**Pull request merged**